blog/2024-02-01-py-client-v4-release/index.mdx

```diff
@@ -0,0 +1,436 @@
+            text={PythonCode}
+            startMarker="# START AggregateCountExample"
+            endMarker="# END AggregateCountExample"
+            language="py"
+        />
+        <p><small>This example retrieves a count of objects that match the filter criteria.</small></p><br/>
+    </TabItem>
+</Tabs>
+
+#### Returned data
+
+As mentioned before, the default behavior now is to return all properties (except `blob`s) by default, as well as object uuids.
+
+For less-often used attributes such as cross-references, metadata (previously under `_additional`) and vectors must be requested specifically. This is done through the `return_references`, `return_metadata` and `include_vector` parameters.
+
+Note in the example below how the `MetadataQuery` and `QueryReference` classes reduce the amount of syntax memorization required to request metadata or cross-referenced data.
+
+<Tabs groupId="languages">
+    <TabItem value="custom" label="Specify returns">
+        <FilteredTextBlock
+            text={PythonCode}
+            startMarker="# START BM25QueryCustomReturnsExample"
+            endMarker="# END BM25QueryCustomReturnsExample"
+            language="py"
+        />
+    </TabItem>
+    <TabItem value="default" label="Default returns">
+        <FilteredTextBlock
+            text={PythonCode}
+            startMarker="# START BM25QueryDefaultReturnsExample"
+            endMarker="# END BM25QueryDefaultReturnsExample"
+            language="py"
+        />
+    </TabItem>
+</Tabs>
+
+#### Responses
+
+Responses from Weaviate are now parsed into instances of custom classes. For example, each query response will have a list of objects in the `objects` attribute, and each object will have a set of attributes itself. Responses for RAG queries will include a `generated` attribute at the top level for `grouped_task` query, and each object will also have a `generated` attribute for `single_prompt` queries.
+
+### Iterator
+
+The client includes an iterator to make convenient use of the `cursor` API. This is useful for retrieving large numbers of objects, such as when manually exporting or migrating a collection.
+
+<Tabs groupId="languages">
+    <TabItem value="basic" label="Default">
+        <FilteredTextBlock
+            text={PythonCode}
+            startMarker="# IteratorBasic"
+            endMarker="# END IteratorBasic"
+            language="py"
+        />
+    </TabItem>
+    <TabItem value="oneprop" label="Selected properties">
+        <FilteredTextBlock
+            text={PythonCode}
+            startMarker="# IteratorAnswerOnly"
+            endMarker="# END IteratorAnswerOnly"
+            language="py"
+        />
+    </TabItem>
+    <TabItem value="meta" label="Metadata Only">
+        <FilteredTextBlock
+            text={PythonCode}
+            startMarker="# IteratorMetadataOnly"
+            endMarker="# END IteratorMetadataOnly"
+            language="py"
+        />
+    </TabItem>
+</Tabs>
```

Comment:
Maybe also mention taht you can do len(collection) ? That is a nice QOL improvement :)

Reply:
Oh I didn't know that! Cool yes

Reply:
Done